go/contacts/resolution_notice.go

```diff
@@ -0,0 +1,143 @@
+package contacts
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"github.com/keybase/client/go/encrypteddb"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/msgpack"
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+type ContactResolution struct {
+	Description  string
+	ResolvedUser keybase1.User
+}
+
+const ContactResolutionEncryptionVersion = 1
+
+type ContactResolutionForEncryption struct {
+	ContactResolution
+	EncryptionVersion int
+}
+
+type EncryptedContactResolution struct {
+	Blob          []byte
+	PukGeneration keybase1.PerUserKeyGeneration
+}
+
+var errWrongContactEncryptionScheme = errors.New(
+	"wrong encryption version for encrypted contact blob")
+
+func getPerUserKeyFn(mctx libkb.MetaContext,
+	pukGen *keybase1.PerUserKeyGeneration) (keyFn func(ctx context.
+	Context) ([32]byte, error), gen keybase1.PerUserKeyGeneration,
+	errOuter error) {
+	pukring, err := mctx.G().GetPerUserKeyring(mctx.Ctx())
+	if err != nil {
+		return nil, gen, err
+	}
+	err = pukring.Sync(mctx)
+	if err != nil {
+		return nil, gen, err
+	}
+	currentGen := pukring.CurrentGeneration()
+	if pukGen == nil {
+		pukGen = &currentGen
+	}
+	keyFn = func(ctx context.Context) (res [32]byte, err error) {
+		keypair, err := pukring.GetEncryptionKeyByGeneration(mctx, *pukGen)
+
+		// Derive symmetric key from PUK
+		skey, err := keypair.SecretSymmetricKey(libkb.EncryptionReasonContactsResolvedServer)
+		if err != nil {
+			return res, err
+		}
+
+		copy(res[:], skey[:])
+		return res, nil
+	}
+	return keyFn, *pukGen, nil
+}
+
+func encryptContactBlob(mctx libkb.MetaContext, res ContactResolution) (string,
```

Comment:
really curious about this style where the error is on the next line lol

Reply:
is it to enforce line length?

Reply:
yeah I have my editor set to roll things to keep lines at 80 chars since the KBFS codebase cares about that. I'm pretty ambivalent to it as long as things are like <= 120ish